### ISSUE TITLE
Integrate linter

### ISSUE THREAD

**Issue by larryonoff** [awslabs/aws-mobile-appsync-sdk-ios#107]
I have suggestion on adding SwiftLint linter to the project, e.g. rules that I'm using in my projects.

```yaml
disabled_rules: # rule identifiers to exclude from running
  - file_length
  - force_cast
  - function_body_length
  - function_parameter_count
  - large_tuple
  - nesting
  - redundant_optional_initialization
  - type_body_length
  - type_name
  - identifier_name
  - weak_delegate

# configurable rules can be customized from this configuration file
closing_brace: error
comma: error
colon:
  severity: error
empty_count: warning
function_body_length:
  warning: 100
  error: 150
opening_brace: error
return_arrow_whitespace: error
statement_position:
  severity: error
todo: warning
trailing_semicolon: error
```

**Comment by palpatim** [comment]
Thanks for this suggestion. We have an internal task to do this as well (it's a pet peeve of mine to not have linted code :) )

We'll be sure to evaluate using your suggested rules above.

**Comment by palpatim** [comment]
Merged #121, thanks!
